serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java

```diff
@@ -803,6 +817,7 @@ public static final class DerProperty<B, P> {
                 .orElse(null);
             this.explicitlyRequired = annotationMetadata.booleanValue(SerdeConfig.class, SerdeConfig.REQUIRED)
                 .orElse(false);
+            this.explicitlyRequiredForConstructor = explicitlyRequired || argument.isPrimitive() && failOnNullForPrimitives;
```

Comment:
```suggestion
            this.explicitlyRequiredForConstructor = explicitlyRequired || (argument.isPrimitive() && failOnNullForPrimitives);
```

Reply:
Why? Having trouble reading it?

Reply:
yes, no idea about the precedence rules for the logical operators

Reply:
Just like in math * / AND has a priority :D